hpke.go

```diff
@@ -136,7 +136,14 @@ type hpkeContext struct {
 	infoHash  []byte `tls:"head=none"`
 }
 
-func keySchedule(suite CipherSuite, mode HPKEMode, pkR KEMPublicKey, zz, enc, info, psk, pskID, pkIm []byte) (key, nonce []byte, err error) {
+type contextParameters struct {
+	context hpkeContext
+	secret []byte
+	key []byte
+	nonce []byte
+}
+
+func keySchedule(suite CipherSuite, mode HPKEMode, pkR KEMPublicKey, zz, enc, info, psk, pskID, pkIm []byte) (params contextParameters, err error) {
```

Comment:
Is there a specific reason you're using named return values here?  Generally, my impression is that they're to be avoided in all but very small functions.

Reply:
No reason other than I don't know how to write idiomatic Go code. :-)